features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp

```diff
@@ -121,10 +194,65 @@ nsapi_error_t UBLOX_AT_CellularContext::open_data_channel()
 
     return err;
 }
+#endif
+
+#ifdef TARGET_UBLOX_C030_R412M
+bool UBLOX_AT_CellularContext::activate_profile(const char *apn,
+                                                const char *username,
+                                                const char *password,
+                                                AuthenticationType auth)
+{
+    bool success = false;
+    int modem_security = nsapi_security_to_modem_security(auth);
+
+    _at.cmd_start("AT+CGDCONT=1,");
+    _at.write_string("IP");
+    _at.write_string(apn);
+    _at.cmd_stop();
+    _at.resp_start();
+    _at.resp_stop();
+
+
+    if (_at.get_last_error() == NSAPI_ERROR_OK) {
+        if (modem_security == CHAP) {
+            _at.cmd_start("AT+UAUTHREQ=1,");
+            _at.write_int(modem_security);
+            _at.write_string(password);
+            _at.write_string(username);
+            _at.cmd_stop();
+            _at.resp_start();
+            _at.resp_stop();
+
+        } else if (modem_security == NOAUTH) {
+            _at.cmd_start("AT+UAUTHREQ=1,");
+            _at.write_int(modem_security);
+            _at.cmd_stop();
+            _at.resp_start();
+            _at.resp_stop();
+
+        } else {
+            _at.cmd_start("AT+UAUTHREQ=1,");
```

Comment:
What options include in else ?
Can this if/else if/ else  be like 
if (modem_security == NOAUTH) {
..
} else {
..
}